mmv1/third_party/cai2hcl/common/hclwrite.go

```diff
@@ -0,0 +1,70 @@
+package common
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/hcl/hcl/printer"
+	"github.com/hashicorp/hcl/v2/hclwrite"
+	"github.com/zclconf/go-cty/cty"
+)
+
+func HclWriteBlocks(blocks []*HCLResourceBlock) ([]byte, error) {
```

Comment:
The HCL & converter_factory extraction is extremely straightforward but ends up causing a decent chunk of the diff and making it harder to track the remaining changes - could you move that to a separate PR that I can approve and merge while we discuss the converter interface?

Reply:
Sure, done: https://github.com/GoogleCloudPlatform/magic-modules/pull/9607